wpe/src/main/java/com/wpe/wpeview/WPECookieManager.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (C) 2022 Igalia S.L. <info@igalia.com>
+ *   Author: Jani Hautakangas <jani@igalia.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+package com.wpe.wpeview;
+
+import android.annotation.SuppressLint;
+import android.os.Handler;
+
+import androidx.annotation.Keep;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public final class WPECookieManager {
+
+    private enum NativeCookieAcceptPolicy {
+        AcceptAlways(0),
+        AcceptNever(1),
+        AcceptNoThirdParty(2);
+
+        private final int value;
+
+        NativeCookieAcceptPolicy(int value) { this.value = value; }
+
+        public int getValue() { return value; }
+    }
+
+    private static WPECookieManager instance = null;
```

Comment:
Here the class is very simple so it will not use much memory, but the singleton getter could be called quite often, so having the singleton built during the class loading is very light and it avoids a hard synchronization point each time somebody needs to access it.

Those pages show the different ways of creating a singleton in Java with the advantages and drawbacks of each approach. I think that here we are clearly in the point 1 situation (eager initialization).

https://www.digitalocean.com/community/tutorials/java-singleton-design-pattern-best-practices-examples
https://www.digitalocean.com/community/tutorials/thread-safety-in-java-singleton-classes

Reply:
fixed